### ISSUE TITLE
[Global] Content above heading

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#765]
## Issue summary

When a screen reader user navigates via headings, any related content "above" the heading in the DOM will be missed. Content above the heading will appear as though its a part of the previous content section, or will be missed altogether.

This is the case with:
- Featured product (vendor content)
- Product cards (image and badge content)
- Featured collection (image)
- Collage (image)
- Multicolumn (image)
- Image with text (image)
- Collection list (image)

This content appears above the heading text which leaves the screen reader user to believe the current item content (vendor, sale/sold out badge, image) belongs to the previous item.

#### HTML

```html
<!-- Featured product example… -->

<div class="card-information__wrapper">
  <span class="visually-hidden">Vendor:</span>
  <div class="caption-with-letter-spacing light">United By Blue</div>
  <h3 class="card-information__text h5">
    <a href="/products/5-panel-hat" class="full-unstyled-link">
      5 Panel Camp Cap
    </a>
  </h3>
  <div class="price">
    <!-- … -->
  </div>
  <!-- … -->
</div>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. With an active screen reader, navigate through the content via headings alone

### Expected

- Related item content to be placed below the heading in the DOM.

### Actual

- Some content is placed above the heading in the DOM.

## Recommendation

1. Place all related item content **below** the heading element.
2. Use CSS to visually position the heading element "after" the content as to not disrupt the intended design.

Ideally the DOM order would be:
1. Heading
2. Card content (meta data, pricing, etc)
3. Image, anything else

The reason this ordering is ideal is when a screen reader user navigates via headings, content after the selected heading will be associated with said heading.

**Note:** This would **not** be a violation of [2.4.3 Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) due to the content in question not being focusable elements. Also, the close proximity in which the content is reversed would have little negative affect when navigating via virtual/browse mode.

#### HTML

```html
<!-- Featured product example… -->

<div class="card-information__wrapper">
  <h3 class="card-information__text h5">
    <a href="/products/5-panel-hat" class="full-unstyled-link">
      5 Panel Camp Cap
    </a>
  </h3>
  <span class="visually-hidden">Vendor:</span>
  <div class="caption-with-letter-spacing light">United By Blue</div>
  <div class="price">
    <!-- … -->
  </div>
  <!-- … -->
</div>
```

#### CSS

```css
/* Featured product example… */

.card-information__wrapper {
  /* … */
  display: flex;
  flex-direction: column;
}

.card-information__text {
  /* … */
  order: 1;
}

.caption-with-letter-spacing {
  /* … */
  order: 0;
}

.price {
  /* … */
  order: 3;
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.2.0
- **Component:** Global
- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.3.2 Meaningful Sequence](https://www.w3.org/WAI/WCAG21/Understanding/meaningful-sequence.html)
- **Severity:** High
- **Effort:** Medium
- **Related:** #450
